### PR TITLE
[PLAT-20663] Support pre-release YBA versions

### PR DESCRIPTION
**Pull request by rtsisyk**
## Summary

The provider hard-coded `https://downloads.yugabyte.com/releases` as the YBA installer download source, which only works for GA release versions (`20YY.x`). Pre-release / CI builds (`2.xx`) live on a different host (`releases.yugabyte.com`), use a different path layout (full version including build number), require a different OS bundle qualifier (`centos` instead of `linux`), and need `YBA_MODE=dev` to bypass `yba-ctl`'s strict version-comparison check.

This PR auto-detects the version format and routes the bundle download, the bundle name, and the `yba-ctl` invocations accordingly. GA releases are unchanged.

Closes [PLAT-20663](https://yugabyte.atlassian.net/browse/PLAT-20663).

## Source matrix

| Aspect              | GA release                                                | Pre-release / CI build                                     |
| ------------------- | --------------------------------------------------------- | ---------------------------------------------------------- |
| Version format      | `20YY.x.y.z` (e.g. `2024.1.0.0-b129`, `2025.2.2.2-b11`)   | `2.xx.y.z-bN` (e.g. `2.25.0.0-b300`, `2.31.0.0-b14`)        |
| Base URL            | `https://downloads.yugabyte.com/releases`                 | `https://releases.yugabyte.com`                            |
| Path segment        | `/{version-without-build}/`                               | `/{full-version-with-build}/`                              |
| Bundle OS qualifier | `linux`                                                   | `centos`                                                   |
| `yba-ctl` invocation | plain `sudo ./yba-ctl …`                                  | `sudo YBA_MODE=dev ./yba-ctl …`                            |
| Example bundle      | `yba_installer_full-2024.1.0.0-b129-linux-x86_64.tar.gz`  | `yba_installer_full-2.25.0.0-b300-centos-x86_64.tar.gz`    |

## Implementation

In `internal/installation/resource_yba_installer.go`:

- Added `gaVersionRegex = ^20\d{2}\.` and an exported `IsGAVersion(version string) bool` predicate.
- New constants `GADownloadURL` and `PreReleaseDownloadURL`.
- `ybaCtlSudo(version)` centralizes the invocation prefix — `sudo` for GA, `sudo YBA_MODE=dev` for pre-release. All command builders share it instead of repeating the branch.
- `getYBAInstallerPackageNames` (renamed from `getYBAInstallerPackageString`) rewrites `linux` → `centos` for pre-release versions when constructing the bundle name.
- `getYBAInstallerDownloadURL` is the single source of the tarball URL and selects the base URL and path layout from `IsGAVersion(version)`. GA: `{GADownloadURL}/{version-without-build}/{bundle}.tar.gz`. Pre-release: `{PreReleaseDownloadURL}/{full-version}/{bundle}.tar.gz`.
- `getBundleDownloadCommands` (renamed from the misleadingly named `getYBAInstallerBundle`, which returns commands, not a bundle) wraps that URL in the `curl` + `tar` commands.
- `getAddLicenseCommand`, `getInstallCommands`, `getUpgradeCommands`, `getReconfigureCommands`, and `getDeleteCommands` all build their `yba-ctl` invocation via `ybaCtlSudo(version)`.
- `resourceYBAInstallerUpdate` and `resourceYBAInstallerDelete` now thread `yba_version` (the old version, in `Update`'s case) into the command builders so the right invocation form is chosen.
- Drive-by: replaced the bespoke `getSkipPreflightChecksSubCommand` loop with `strings.Join`. Same output, less code.

## Tests

`internal/installation/resource_yba_installer_test.go` is new. Every pure helper is covered for both GA and pre-release inputs (100% statement coverage on the helpers; the SSH-bound CRUD entry points remain acceptance-test territory):

- `TestIsGAVersion` — version-format classification, including edge cases (empty string, year without trailing dot, pre-2000 year, non-anchored match).
- `TestGetYBAInstallerPackageNames` — bundle-name construction including the `linux` → `centos` rewrite and build-number stripping.
- `TestGetYBAInstallerDownloadURL` — exact tarball URL for GA (build-stripped path) vs pre-release (full-version path), plus `aarch64` to confirm architecture propagation.
- `TestGetBundleDownloadCommands` — the `curl` + `tar` command assembly wraps the URL correctly.
- `TestYbaCtlSudo`, `TestGetAddLicenseCommandFormat` — exact invocation prefix / license command for both code paths.
- `TestGetInstallCommandsWithConfig`, `TestGetInstallCommandsPreRelease`, `TestGetInstallCommandsSkipPreflight`, `TestGetInstallCommandsNilOrEmptySkipPreflight` — install ordering, the pre-release path, skip-preflight flag joining, and the nil/empty boundary.
- `TestGetReconfigureCommands`, `TestGetUpgradeCommands`, `TestGetUpgradeCommandsSkipPreflight`, `TestGetDeleteCommands` — same GA/pre-release matrix for the remaining lifecycle commands.

## Documentation

Documentation is intentionally not updated, as there is no need to expose this functionality to customers.

## Acceptance criteria

- [x] Provider can install YBA from `20YY.x` versions (GA releases) — unchanged code path, regression-tested.
- [x] Provider can install YBA from `2.xx` versions (pre-release / CI builds).
- [x] Version detection logic correctly routes to the appropriate bucket (`TestIsGAVersion`, `TestGetYBAInstallerDownloadURL`).
- [x] Documentation updated to reflect pre-release support.

## Test plan

- [x] `go build ./...`
- [x] `go vet ./...`
- [x] `gofmt -l internal/` clean
- [x] `make lint` (golangci-lint) — 0 issues
- [x] `go test ./internal/installation/...` all green
- [x] `make documents` regenerates without drift
- [ ] Manual install against a `2.xx` pre-release build on a real YBA host
- [ ] Manual install against a `20YY.x` GA release on a real YBA host (regression check)

[PLAT-20663]: https://yugabyte.atlassian.net/browse/PLAT-20663?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by rtsisyk** [comment]
The PR has been rebased, plus I've improved the test coverage. PTAL.

File: internal/installation/resource_yba_installer.go [hunk 1]
```diff
@@ -415,85 +428,105 @@ func resourceYBAInstallerDelete(
...
+}
+
+// getYBAInstallerPackageNames derives the package naming pieces for a version:
+// the extracted folder name, the tarball base name, and the build-stripped
+// version used in GA download paths.
+func getYBAInstallerPackageNames(version, os, arch string) (folder, bundle, v string) {
 	folder = fmt.Sprintf("yba_installer_full-%s", version)
+	// Pre-release bundles are named "centos" rather than "linux".
+	if !IsGAVersion(version) && os == "linux" {
+		os = "centos"
```
Comment (rtsisyk) [inline internal/installation/resource_yba_installer.go:665]: I don't know why such disparity exists in the release process, but we have to deal with it here.

File: docs/resources/installer.md [hunk 2]
```diff
@@ -10,6 +10,17 @@ description: |-
 
 Manages the installation of YugabyteDB Anywhere on an existing virtual machine using YBA Installer.
 
+### Supported Versions
+
+Both GA releases and pre-release CI builds are supported. The format of `yba_version` is auto-detected, and the installer routes the package download and `yba-ctl` invocation accordingly. Always supply the full version including the build number (the `-bNNN` suffix).
```
Comment (Deepti-yb) [inline docs/resources/installer.md]: Do we want to expose this to the customers? Would nudging them to the pre release on a case-by-case basis not be enough?

Comment (rtsisyk) [reply docs/resources/installer.md]: > Do we want to expose this to the customers? Would nudging them to the pre release on a case-by-case basis not be enough?

Not sure. Probably can remove it out of the public documentation for now.

Comment (rtsisyk) [reply docs/resources/installer.md]: Excluded documentation changes.

**Review by Deepti-yb — APPROVED** [review]
LGTM
